### DISCUSSION TITLE
Authentication failed - How can I clear and re-add SSH

### DISCUSSION THREAD

**Discussion by jordankeen** [community/community#64661] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

Hey, so I'm stuck with the below error when I try to push in github desktop. I have tried several things through the docs to make a new SSH key but have had no luck.

Is there a simple way for me to clear all in my host file, clear all ssh keys, and start over?

_Authentication failed. Some common reasons include:

- You are not logged in to your account: see GitHub Desktop > Settings.
- You may need to log out and log back in to refresh your token.
- You do not have permission to access this repository.
- The repository is archived on GitHub. Check the repository settings to confirm you are still permitted to push commits.
- If you use SSH authentication, check that your key is added to the ssh-agent and associated with your account.
- If you use SSH authentication, ensure the host key verification passes for your repository hosting service.
- If you used username / password authentication, you might need to use a Personal Access Token instead of your account password. Check the documentation of your repository hosting service._

**Comment by bronglil** [comment]
@jordankeen Of course! It seems like you're encountering authentication issues when trying to push to GitHub using GitHub Desktop. Here's a step-by-step guide to troubleshoot and resolve the issue if these steps work kindly mark this as the correct answer thanks 

Check GitHub Desktop Settings:

Open GitHub Desktop.
Click on the "File" menu (on macOS) or the "Edit" menu (on Windows), and select "Options" or "Preferences."
In the "Options" or "Preferences" window, go to the "Accounts" tab and make sure you're logged in to the correct GitHub account.
Refresh Your Token:

If you're logged in but still facing issues, try logging out and then logging back in. This will refresh your authentication token.
Go to GitHub Desktop > File (Edit on Windows) > Options (Preferences on macOS) > Accounts.
Click on the account you're using and select "Sign Out."
Log back in with your GitHub credentials.
SSH Key Verification:

If you're using SSH authentication, ensure that your SSH key is correctly added to the ssh-agent and associated with your GitHub account:
Open a terminal.
Run the command ssh-add -l to list the currently loaded keys. If your key isn't listed, add it using ssh-add path/to/your/private/key.
Make sure the SSH key is added to your GitHub account settings. You can copy the public key content (cat ~/.ssh/id_rsa.pub) and add it to your GitHub account's SSH settings.

**Reply by bronglil** [reply]
or you can try create a ssh key on local machine and paste it on the github

**Comment by nirav-gajera** [comment, marked as answer]
To clear all in your host file and all SSH keys, you can follow these steps:

1. Open a terminal window.
2. Run the following command to clear your host file:

```
sudo rm -f ~/.ssh/known_hosts
```

3. Run the following command to clear all SSH keys:

```
ssh-keygen -f ~/.ssh/id_rsa -e
```

This will delete all SSH keys that are stored in your `~/.ssh` directory.

-To re-add your SSH key to GitHub:


1. Open a terminal window.
2. Run the following command to generate a new SSH key:

```
ssh-keygen -t rsa -b 4096 -C "your_email@example.com"
```

This will create a new SSH key pair in your `~/.ssh` directory. The default filename for the private key is `id_rsa` and the default filename for the public key is `id_rsa.pub`.

3. When prompted, enter a passphrase for your SSH key. This passphrase will be used to protect your private key.
4. Once you have generated your SSH key, copy the contents of the public key to the clipboard.

5. Go to your GitHub account settings and click on "SSH and GPG keys".
6. Click on the "New SSH key" button.
7. Paste the contents of your public key into the "Key" field.
8. Give your key a title and click on the "Create SSH key" button.

**Reply by jordankeen** [reply]
@nirav-gajera Hey thank you for your help! This seemed to work, I did get a small error that popped up when I tried to push from my github desktop app but after that it seems to be back to normal

**Comment by jasonwcc** [comment]
I even get this error thru github.com online gui way. why this ? 
![image](https://github.com/user-attachments/assets/ea5e8843-2cd8-4787-847b-07b7021a6760)

![Uploading github error1.png…]()

**Reply by benrich37** [reply]
For anyone else brought here by this error, this was caused for me by accidentally opening the wrong 2FA app (so the error was caused by typing the wrong 2FA code)
